web-wallet/src/lib/components/ApproveMigration/ApproveMigration.svelte

```diff
@@ -0,0 +1,143 @@
+<svelte:options immutable={true} />
+
+<script>
+  import { mdiAlertOutline, mdiTimerSand } from "@mdi/js";
+  import { getAccount, waitForTransactionReceipt } from "@wagmi/core";
+  import { isHex } from "viem";
+  import { createEventDispatcher } from "svelte";
+  import { Button, Icon } from "$lib/dusk/components";
+  import { wagmiConfig } from "$lib/migration/walletConnection";
+  import { allowance, approve } from "$lib/migration/migration";
+  import { createDataStore } from "$lib/dusk/svelte-stores";
+
+  /** @type {number} */
+  export let amount;
+
+  /** @type {`0x${string}`} */
+  export let chainContract;
+
+  /** @type {`0x${string}`} */
+  export let migrationContract;
+
+  /** @type {import("svelte/store").Writable<Boolean>}*/
+  export let hasApprovedCoin;
+
+  const dispatch = createEventDispatcher();
+
+  const approvalTxStore = createDataStore(handleApprove);
+
+  const retrieveAccount = () =>
+    new Promise(() => {
+      return getAccount(wagmiConfig);
+    });
+
+  $: ({ isLoading, data, error } = $approvalTxStore);
+
+  /**
+   * @param { `0x${string}` } address
+   */
```

Comment:
A nitpick, but let's perhaps have this as one liner (ie. `/** @param { `0x${string}` } address */`), so that it matches the format of the type definitions above.

I don't think we have some guidance specified around the format we use, but I think we are sometimes a bit inconsistent with it. Same goes with uppercasing the types (ie. `Number` vs `number`). I think it would be good to settle on a standard around these.